### DISCUSSION TITLE
Does SWS have some caching mechanisms?

### DISCUSSION THREAD

**Discussion by Relaxe111** [static-web-server/static-web-server#241] [Q&A]
Hi, i have managed to create a cdn connected to s3 storage with help of `https://github.com/yandex-cloud/k8s-csi-s3`. Now I can serve static files from remote s3 bucket. 
Time for serving index.html is around 200 ms. 3 mb around 3 secs. 
Is there a way of caching static resources somewhere locally or in memory such that it will fetch files from s3 only on first request but all subsequent requests from local cache?

**Comment by joseluisq** [comment]
No such cache mechanism like in memory or local cache, maybe in the future.

But how about browser cache like [`cache-control`](https://static-web-server.net/features/cache-control-headers/)?
You can also customize those headers for particular files with [custom headers](https://static-web-server.net/features/custom-http-headers/)

**Comment by Relaxe111** [comment]
Yes that I have done cache-control. But i think i can get more performance if one will have some of `caching` techniques on server level. 

Is there any issue opened related to `memory or local cache`?  Maybe to open one?

I have no experience with Rust last years (neither with, SWS) but i could try to find an solution and do an pr if someone can help to tell where to start.

**Comment by Relaxe111** [comment]
I see that Rust has something quite interesting implementations for in memory file caching. [https://docs.rs/freqfs/0.4.3/freqfs/](https://docs.rs/freqfs/0.4.3/freqfs/)

**Comment by joseluisq** [comment]
Sure, feel free to open an issue if you thinks so.
You could start describing the problem or use case and suggest something.

From my side, I can try to help you along the way.

**Comment by Relaxe111** [comment]
have created one [242](https://github.com/static-web-server/static-web-server/issues/242)

**Comment by Relaxe111** [comment]
closing
